js/ui/workspaceMonitor.js

```diff
@@ -0,0 +1,68 @@
+// -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
+
+const Lang = imports.lang;
+const Shell = imports.gi.Shell;
+
+const Main = imports.ui.main;
+
+const WorkspaceMonitor = new Lang.Class({
+    Name: 'WorkspaceMonitor',
+
+    _init: function() {
+        this._shellwm = global.window_manager;
+        this._shellwm.connect('minimize-completed', Lang.bind(this, this._updateOverview));
+        this._shellwm.connect('destroy-completed', Lang.bind(this, this._updateOverview));
+
+        this._metaScreen = global.screen;
+        this._metaScreen.connect('in-fullscreen-changed', Lang.bind(this, this._fullscreenChanged));
+
+        let primaryMonitor = Main.layoutManager.primaryMonitor;
+        this._inFullscreen = primaryMonitor && primaryMonitor.inFullscreen;
+
+        this._appSystem = Shell.AppSystem.get_default();
```

Comment:
I'm not sure that's better (I kind of like calling the getter once only for the lifetime of workspaceMonitor), but ok